teps/0100-embedded-taskruns-and-runs-status-in-pipelineruns.md

```diff
@@ -171,12 +195,389 @@ In addition to grouping related resources, Tekton Results provides long term sto
+is backwards incompatible. 
+
+To support migration as required by our [API compatibility policy][deprecations]
+we will add a behavior flag - `embedded-status` - used to configure whether the
+`PipelineRuns` should contain the full embedded status or the minimal embedded
+status of its `TaskRuns` and `Runs`.
+
+Following our [policy][behavior-flags] on updating behavior flags:
+1. The `embedded-status` flag will be `full` by default, users can set it to `minimal`. 
+2. After 9 months in v1beta1, the `embedded-status` flag will be changed to `minimal`
+  by default.
+3. As soon as the next release in v1beta1, `embedded-status` flag will be removed as
+  well as the full embedded status fields.
+
+###### Alternatives
+* [Use Booleans](#beta-api---use-booleans)
+* [Provide Both Full and Minimal Statuses](#beta-api---provide-both-full-and-minimal-statuses)
+
+### V1 API
+
+In V1, we will have minimal embedded statuses of `TaskRuns` and `Runs` in `PipelineRuns`.
+
+```go
+type PipelineRunStatusFields struct {
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
+	TaskRunsStatuses []PipelineRunTaskRunMinimalStatus `json:"taskRunsStatuses,omitempty"`
+	RunsStatuses []PipelineRunRunMinimalStatus `json:"runsStatuses,omitempty"`
+	PipelineResults []PipelineRunResult `json:"pipelineResults,omitempty"`
+	PipelineSpec *PipelineSpec `json:"pipelineSpec,omitempty"`
+	SkippedTasks []SkippedTask `json:"skippedTasks,omitempty"`
+}
+```
+The full embedded statuses of `TaskRuns` and `Runs` will not be available in `PipelineRuns`.
+Read more about V1 in [TEP-0096: Pipelines V1 API][tep-0096].
+
+### JSON keys
+
+The JSON struct tags defined on Kubernetes CRDs are used to serialize and deserialize
+instances of those CRDs (see [JSON and Go][json-and-go] for more information on how
+struct tags affect serialization and deserialization). We can't reuse the existing JSON
+keys `taskRuns` and `runs` without breaking serialization/deserialization for existing
+`PipelineRuns`. The keys can be reused once we move to a V1 API by writing a conversion
+webhook, but it is easier to simply not reuse the keys.
+
+### Tekton Projects
+
+#### Tekton Results
+
+As described in the [background section](#tekton-results-api), the Results API enables
+users to bundle `TaskRuns` and `Runs` to their parent `PipelineRuns`. It also provides
+long term storage of resources. Users can rely on [Tekton Results][results-api] to map
+`TaskRuns` and `Runs` to provide the mapping that was available in the full embedded
+statuses. 
+
+Note that Results API is still in Alpha, but progress is being made towards Beta - we
+estimate that the Results API will be in Beta by the time we remove the full embedded
+statuses.
+
+#### Tekton Dashboard
+
+[Tekton Dashboard][dashboard] shows the status the `TaskRuns` and `Runs` of a given
+`PipelineRun`, this should be unaffected. We expect the load on the Dashboard to
```

Comment:
> Tekton Dashboard… this should be unaffected

Not entirely true. Some changes will be needed in the Dashboard as it currently relies on the `status.taskRuns` field in a few places. We can update these to use the new field, but will need to be careful to ensure a smooth transition.

Reply:
the "this should be unaffected" meant more of "this should continue to be supported" - clarified this in the TEP

sounds great, we'll have 9 months to do the migrations - will reach out to you and the dashboard team when this feature is released so we can start migrating :)